Aztec/Classes/TextKit/TextView.swift

```diff
@@ -527,41 +583,142 @@ open class TextView: UITextView {
+
+        } else {
+            insertText(String(.tab))
+        }
+        
+    }
+    
+    
+    // MARK: - Text List indent methods
+    
+    private func indent(list: TextList, increase: Bool = true) {
 
         let formatter = TextListFormatter(style: list.style, placeholderAttributes: nil, increaseDepth: true)
-        let liFormatter = LiFormatter(placeholderAttributes: nil)
+        let li = LiFormatter(placeholderAttributes: nil)
+
         let targetRange = formatter.applicationRange(for: selectedRange, in: storage)
 
         performUndoable(at: targetRange) {
-            let finalRange = formatter.removeAttributes(from: storage, at: targetRange)
-            liFormatter.removeAttributes(from: storage, at: targetRange)
+            let finalRange: NSRange
+            if increase {
+                finalRange = increaseIndent(listFormatter: formatter, liFormatter: li, targetRange: targetRange)
+            } else {
+                finalRange = decreaseIndent(listFormatter: formatter, liFormatter: li, targetRange: targetRange)
+            }
             typingAttributes = textStorage.attributes(at: targetRange.location, effectiveRange: nil)
             return finalRange
         }
-    }
 
-    @objc func handleTab(command: UIKeyCommand) {
-        let lists = TextListFormatter.lists(in: typingAttributes)
-        guard let list = lists.last, lists.count < maximumListIndentationLevels else {
-            insertText(String(.tab))
-            return
-        }
-
-        let formatter = TextListFormatter(style: list.style, placeholderAttributes: nil, increaseDepth: true)
-        let liFormatter = LiFormatter(placeholderAttributes: nil)
+    }
+    
+    // MARK: Text List increase or decrease indentation
+    private func increaseIndent(listFormatter: TextListFormatter, liFormatter: LiFormatter, targetRange: NSRange) -> NSRange {
+        let finalRange = listFormatter.applyAttributes(to: storage, at: targetRange)
+        liFormatter.applyAttributes(to: storage, at: targetRange)
+        
+        return finalRange
+    }
+    
+    private func decreaseIndent(listFormatter: TextListFormatter, liFormatter: LiFormatter, targetRange: NSRange) -> NSRange {
+        let finalRange = listFormatter.removeAttributes(from: storage, at: targetRange)
+        liFormatter.removeAttributes(from: storage, at: targetRange)
+        
+        return finalRange
+    }
+    
+    
+    // MARK: - Blockquote indent methods
+    
+    private func indent(blockquote: Blockquote, increase: Bool = true) {
+        
+        let paragraphStyle = typingAttributes.paragraphStyle()
+        let currentDepth = paragraphStyle.blockquoteNestedIndent.depth
+        
+        let textColor = getTextColorForQuoteDepth(currentDepth)
+                
+        var previousAttributes = typingAttributes
+        previousAttributes.removeValue(forKey: .foregroundColor)
+        previousAttributes[.foregroundColor] = textColor.prev
```

Comment:
I don't think we should change the color of text in the quote, only the border/bar.

This could be a future nice to have, but will bring us problems with color attributes in the text implemented by style attributes and also can bring issues with dark mode support.

Even if we decide to go ahead and do this, the change of attributes needs to be done inside the formatter class not in this location.

Reply:
Hmm, I really need this feature though. My project which uses aztec needs the nested quoting to look like the apple mail app, which changes text color rather than the background color. Can you point me in the way of doing this in the formatter class? I will take a look myself now that you mention it as well.